doc/source/example-mxnet-walk-through.rst

```diff
@@ -0,0 +1,349 @@
+            data = gluon.utils.split_and_load(batch.data[0], ctx_list=device, batch_axis=0)
+            # Splits validation label into multiple slices along batch_axis
+            # and copy each slice into a context.
+            label = gluon.utils.split_and_load(
+                batch.label[0], ctx_list=device, batch_axis=0
+            )
+            outputs = []
+            for x in data:
+                outputs.append(net(x).softmax())
+            # Updates internal evaluation
+            metrics.update(label, outputs)
+        return metrics.get_name_value()
+
+Having defined defining the data loading, model architecture, training, and evaluation we can put everything together and train our model on MNIST. Note that the GPU/CPU device for the training and testing is defined within the :code:`ctx` (context).  
+
+.. code-block:: python
+
+    def main():
+        # Setup context to GPU and if not available to CPU
+        DEVICE = [mx.gpu() if mx.test_utils.list_gpus() else mx.cpu()]
+        # Load train and validation data
+        train_data, val_data = load_data()
+        # Define sequential model
+        net = model()
+        init = nd.random.uniform(shape=(2, 784))
+        net(init)
+        # Start model training based on training set
+        train(net=net, train_data=train_data, epoch=5, device=DEVICE)
+        # Evaluate model using loss and accuracy
+        eval_metric = test(net=net, val_data=val_data, device=DEVICE)
+        acc = eval_metric[0]
+        loss = eval_metric[1]
+        print("Evaluation Loss: ", loss)
+        print("Evaluation Accuracy: ", acc)
+
+    if __name__ == "__main__":
+            main()
+
+You can now run your MXNet machine learning workload:
+
+.. code-block:: python
+
+    python3 mxnet_mnist.py
+
+So far this should all look fairly familiar if you've used MXNet or even PyTorch before.
+Let's take the next step and use what we've built to create a simple federated learning system consisting of one server and two clients.
+
+MXNet meets Flower
+------------------
+
+So far it was not possible to train a MXNet workload federated since it was not supported yet. However, Flower is fully agnostic to machine learning frameworks and can take any model parameters and aggregate them. This section will show you how Flower can be used to federate MXNet workloads.
+
+The concept is however always the same. 
+We have to start a *server* and then use the code in :code:`mxnet_mnist.py` for the *clients* that are connected to the *server*.
+The *server* sends model parameters to the clients. The *clients* run the training and update the paramters.
+The updated parameters are sent back to the *server* which averages all received parameter updates.
+This describes one round of the federated learning process and we repeat this for multiple rounds.
+The challenging part is to transform the MXNet parameters from :code:`NDArray` to :code:`NumPy Arrays` to make it readable for Flower. 
```

Comment:
Move this sentence closer to where we actually implement this, perhaps after line 251?

Reply:
done